client/src/components/AddProduct.js

```diff
@@ -0,0 +1,349 @@
+import React, { Component } from "react";
+import axios from "axios";
+import DateFnsUtils from "@date-io/date-fns";
+import {
+  MuiPickersUtilsProvider,
+  KeyboardDatePicker
+} from "@material-ui/pickers";
+import { makeStyles } from "@material-ui/core/styles";
+import {
+  TextField,
+  Button,
+  FormControl,
+  Select,
+  MenuItem,
+  Grid
+} from "@material-ui/core";
+
+export default class AddProduct extends Component {
+  state = {
+    title: "",
+    description: "",
+    imageUrl: [],
+    brand: "",
+    category: "",
+    quantity: 0,
+    price: 0,
+    currency: "",
+    tags: [],
+    company: "",
+    location: {
+      latitude: null,
+      longitude: null
+    },
+    // Must be future date
+    availability: new Date(),
+    warrantyUntil: new Date(),
+    condition: ""
+  };
+
+  handleSubmit = event => {
+    event.preventDefault();
+    console.log("handlesubmit working");
+    axios
+      .post("/api/products", {
+        title: this.state.title,
+        description: this.state.description,
+        imageUrl: this.state.imageUrl,
+        brand: this.state.brand,
+        category: this.state.category,
+        quantity: this.state.quantity,
+        price: this.state.price,
+        currency: this.state.currency,
+        company: this.state.company,
+        location: {
+          latitude: this.state.latitude,
```

Comment:
not sure if lat long is just this.state.latitude or this.state.location.latitude. Just a comment that this needs to be verified once the GoogleMap input it setup